azure-functions/acbs-function/src/functions/activity-update-facility-covenant.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Facility covenant amendment DAF
+ * ***************************
+ * This DAF (Durable Activity Function) is never invoked directly.
+ * It is invoked via DOF (Durable Orchestrator Function).
+ *
+ * Pre-requisites
+ * --------------
+ * 0. 'npm install durable-functions'
+ * 1. Durable  HTTP trigger function (acbs-http)
+ * 2. Durable Orchestrator function (DOF) (acbs-amend-facility)
+ *
+ * ------------------
+ * HTTP -> DOF -> DAF
+ * ------------------
+ */
+const df = require('durable-functions');
+const { getNowAsIsoString } = require('../../helpers/date');
+const api = require('../../api');
+const { isHttpErrorStatus } = require('../../helpers/http');
+
+/**
+ * This function is used to update the covenant of a facility. It first checks if the payload is valid and contains all mandatory fields.
+ * If the payload is valid, it sends a request to the API to update the facility covenant.
+ * If the API request is successful, it returns an object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
+ * If the API request fails, it throws an error with details about the request and the error.
+ * If the payload is not valid or does not contain all mandatory fields, it returns an object with the missing mandatory fields.
+ * If any other error occurs, it logs the error and throws a new error.
+ *
+ * @param {Object} payload - The payload containing the facilityIdentifier and acbsFacilityCovenantInput.
+ * @param {string} payload.facilityIdentifier - The identifier of the facility.
```

Comment:
Param needs to be updated here and on other please.

Reply:
What part needs to be updated sorry?